RoboJackets/igvc-software
# Feat/recovery behavior

# Description

Implements a `back_up_recovery` recovery behavior, which makes the robot back up when it gets stuck.

This PR does the following:
- Adds `back_up_recovery` 
- Sets to it be the default recovery behavior for `navigation_server`

Fixes #597 and Fixes #632 

# Testing steps (If relevant)
## Test Case 1
1. Launch a simulation using `roslaunch igvc_gazebo autonav.launch`
2. Set the robot moving using `roslaunch igvc_naviation navigation_simulation.launch`

Expectation: The robot should attempt to turn around, get stuck on the wall, back up, and then turn around again and move forward.

# Self Checklist
